project-service/src/app.js

```diff
@@ -0,0 +1,36 @@
+import express from 'express';
+import morgan from 'morgan';
+import bodyParser from 'body-parser';
+
+import auth from './routes/auth';
+import users from './routes/users';
+import projects, { validation } from './routes/projects';
+import sprints from './routes/sprints';
+import tasks from './routes/tasks';
+
+let app = express();
+
+// middleware for http request logging
+app.use(morgan('dev'));
+
+// middlewares will parse incoming request bodies and populate req.body
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(bodyParser.json());
+
+// routes
+app.use(auth(
+  process.env.AUTH0_SECRET || 'secret',
+  process.env.AUTH0_AUDIENCE || 'audience'
+));
+app.use(users); // maintains user table using Auth0 JSON Web Token
+app.param('projectId', validation); // stores project in req.project if valid
+app.use('/projects/:projectId/tasks', tasks);
+app.use('/projects/:projectId/sprints', sprints);
+app.use('/projects', projects);
```

Comment:
i'm a little surprised that you are right - how come /projects/:projectId/sprints for example won't get routed to projects?